### PR TITLE
Instrument/dynamic handlers

### PR DESCRIPTION
**Pull request by chrzaszcz**
This PR allows dynamic modification of the configured handlers, which could be useful in tests, similarly to `mongoose_modules` and `mongoose_service`.

Previously, the tests used `meck` to inject logging events to an ETS table. Now it is possible to do this with `add_handler` and `remove_handler`.

Calling `intrument_helper:add(my_handler, Opts)` results in:
  - Setting `#{my_handler => Opts}` in the`instrumentation` section of `mongoose_config`.
  - Adding `mongoose_instrument_my_handler` module to handlers.
  - Setting up the new handler for already registered events.

Calling `instrument_helper:remove(my_handler)` does the opposite:
  - Unregister the handler for all events.
  - Remove the options from `mongoose_config`.

By modifying `mongoose_config` in the correct order, there is consistency between the configuration and the registered handlers.

### CODE REVIEW

**Comment by mongoose-im** [comment]
[elasticsearch_and_cassandra_26](https://circleci.com/gh/esl/MongooseIM/213031) / elasticsearch_and_cassandra_mnesia / 429b944c932f1b8921baffba1039b950769deab5
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/4247/213031/elasticsearch_and_cassandra_mnesia.26.1.2-amd64)/ [big](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/4247/213031/elasticsearch_and_cassandra_mnesia.26.1.2-amd64/big/ct_run.test%408f7eaf0b53dc.2024-03-20_18.07.41/index.html)
**OK: 435** / Failed: 0 / **User-skipped: 41** / Auto-skipped: 0

**Comment by mongoose-im** [comment]
[elasticsearch_and_cassandra_26](https://circleci.com/gh/esl/MongooseIM/213102) / elasticsearch_and_cassandra_mnesia / adfa20616a8905708d1c896612e27e6a10bd6067
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/4247/213102/elasticsearch_and_cassandra_mnesia.26.1.2-amd64)/ [big](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/4247/213102/elasticsearch_and_cassandra_mnesia.26.1.2-amd64/big/ct_run.test%4033c786aff01b.2024-03-21_07.35.52/index.html)
**OK: 435** / Failed: 0 / **User-skipped: 41** / Auto-skipped: 0

**Comment by mongoose-im** [comment]
[elasticsearch_and_cassandra_26](https://circleci.com/gh/esl/MongooseIM/213132) / elasticsearch_and_cassandra_mnesia / 3444fe99879164a0b7f802390174a32beffd27ba
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/4247/213132/elasticsearch_and_cassandra_mnesia.26.1.2-amd64)/ [big](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/4247/213132/elasticsearch_and_cassandra_mnesia.26.1.2-amd64/big/ct_run.test%405356b4d5771e.2024-03-21_07.46.05/index.html)
**OK: 435** / Failed: 0 / **User-skipped: 41** / Auto-skipped: 0

**Review by JanuszJakubiec — APPROVED** [review]
Looks good!

File: src/instrument/mongoose_instrument.erl [hunk 1]
```diff
@@ -256,6 +318,10 @@ get_handlers(EventName, Labels) ->
 handle_event(Event, Labels, Measurements, {EventHandlers, Config}) ->
     lists:foreach(fun(Handler) -> Handler(Event, Labels, Config, Measurements) end, EventHandlers).
 
+-spec modules_to_funs([module()]) -> [handler_fun()].
+modules_to_funs(Modules) ->
+    [fun Module:handle_event/4 || Module <- Modules].
+
```
Comment (NelsonVides) [inline src/instrument/mongoose_instrument.erl:324]: Somewhat unrelated to the dynamic handlers but this little thing is a gem to me. Performance!

**Review by NelsonVides — APPROVED** [review]
Love it! 🔥

File: src/instrument/mongoose_instrument.erl (hunk 1 above)
Comment (chrzaszcz) [reply src/instrument/mongoose_instrument.erl:324]: I updated it after your feedback some time ago, and after checking that `gen_*` in OTP do the same.

**Pull request merged**